numpy/__init__.py

```diff
@@ -205,11 +208,35 @@ def pkgload(*packages, **options):
 
     # Make these accessible from numpy name-space
     #  but not imported in from numpy import *
+    #  and make them issue a DeprecationWarning when used
     if sys.version_info[0] >= 3:
-        from builtins import bool, int, float, complex, object, str
-        unicode = str
+        import builtins as _builtins
     else:
-        from __builtin__ import bool, int, float, complex, object, unicode, str
+        import __builtin__ as _builtins
+    for _name, _numpy_equiv in [
+            ("bool", "bool_"),
+            ("int", "int_"),
+            ("float", "float64"),
+            ("complex", "complex128"),
+            ("object", "object_"),
+            ("str", "string_"),
```

Comment:
Not correct - `np.dtype(str)` and `np.dtype(np.string_)` are different on python 3.